src/compas_robots/viewer/scene/robotmodelobject.py

```diff
@@ -52,12 +51,18 @@ def __init__(
         self.hide_coplanaredges = hide_coplanaredges
         self._show_visual = show_visual or True
         self._show_collision = show_collision or False
-        self.configuration: Configuration = configuration or model.zero_configuration()
-        super(RobotModelObject, self).__init__(model=model, **kwargs)
+        self.configuration: Configuration = configuration
+        super(RobotModelObject, self).__init__(**kwargs)
 
         self.visual_objects: list[MeshObject] = self.draw_visual()
         self.collision_objects: list[MeshObject] = self.draw_collision()
 
+    @property
+    def viewer(self):
+        from compas_viewer import Viewer
+
+        return Viewer()
```

Comment:
is there a singleton behind this or will it create a new instance everytime?

Reply:
it's a singleton

Reply:
this is a single tone.